src/api/java/com/minecolonies/api/util/InstantStructurePlacer.java

```diff
@@ -53,16 +51,16 @@ public InstantStructurePlacer(final World worldObj, final String name)
      */
     public static void loadAndPlaceStructureWithRotation(
       final World worldObj, @NotNull final String name,
-      @NotNull final BlockPos pos, final int rotations, @NotNull final Mirror mirror,
-      final boolean complete)
+      @NotNull final BlockPos pos, final Rotation rotation,
+      @NotNull final Mirror mirror, final boolean complete)
     {
         try
         {
-            @NotNull final InstantStructurePlacer structureWrapper = new InstantStructurePlacer(worldObj, name);
-            structureWrapper.structure.setPosition(pos);
-            structureWrapper.rotate(rotations, worldObj, pos, mirror);
-            structureWrapper.structure.setPlacementSettings(new PlacementSettings(mirror, BlockPosUtil.getRotationFromRotations(rotations)));
-            structureWrapper.placeStructure(pos.subtract(structureWrapper.structure.getOffset()), complete);
+        	@NotNull final Structure structure = new Structure(worldObj, name, new PlacementSettings(mirror, rotation));
+            structure.setPosition(pos);
+            structure.rotate(rotation, worldObj, pos, mirror);
+            @NotNull final InstantStructurePlacer structureWrapper = new InstantStructurePlacer(structure);
+            structureWrapper.setupStructurePlacement(pos.subtract(structure.getOffset()), complete, null);
```

Comment:
This no longer instantly executes placeStructure, but instead adds them to some queue, is that ok?